### PR TITLE
Remote queries view using mock data

### PR DESCRIPTION
**Pull request by charisk**
This changeset builds up an initial remote queries view based on some mock data. The view will gradually be refined and hooked into real data with further PRs.

See internal issue for more context.

image

image

## Checklist
N/A: This is only relevant for remote queries, which is still an internal-only feature.

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] `@github/docs-content-codeql` has been cc'd in all issues for UI or other user-facing changes made by this pull request.

### CODE REVIEW

**Review by shati-patel — COMMENTED** [review]
Hooray, thanks for doing this! Looking much nicer already 🖼️ 😄 

image

I've left some minor comments so far, but it would be good to get another review too!

File: extensions/ql-vscode/src/view/octicons.tsx [hunk 1]
```diff
@@ -20,3 +20,19 @@ export const listUnordered = <svg className="octicon octicon-light" width="16" h
 export const info = <svg className="octicon octicon-light" width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg" >
   <path fillRule="evenodd" clipRule="evenodd" d="M8.568 1.03a6.8 6.8 0 0 1 4.192 2.02 7.06 7.06 0 0 1 .46 9.39 6.85 6.85 0 0 1-8.58 1.74 7 7 0 0 1-3.12-3.5 7.12 7.12 0 0 1-.23-4.71 7 7 0 0 1 2.77-3.79 6.8 6.8 0 0 1 4.508-1.15zm.472 12.85a5.89 5.89 0 0 0 3.41-2.07 6.07 6.07 0 0 0-.4-8.06 5.82 5.82 0 0 0-7.43-.74 6.06 6.06 0 0 0 .5 10.29 5.81 5.81 0 0 0 3.92.58zM8.51 7h-1v4h1V7zm0-2h-1v1h1V5z" />
 </svg>;
+
```
[the remark this answers is not in the document]
Comment (charisk) [reply extensions/ql-vscode/src/view/octicons.tsx:23]: Hmm I'm not 100% sure either - I think code and icons have different licenses (MIT vs GitHub license?). See [here](https://github.com/primer/octicons#license).

Perhaps we can avoid mentioning the license here, but agree having a link to where they came from would be useful!

File: extensions/ql-vscode/src/remote-queries/view/RemoteQueries.tsx [hunk 2]
```diff
@@ -1,15 +1,112 @@
...
+};
+
+const data: Props = {
+  queryTitle: 'Empty block',
+  queryFile: 'example.ql',
+  totalRepositoryCount: 13,
+  totalResultCount: 72,
+  executionTimestamp: formatDate(new Date()),
+  executionDuration: '0.6 seconds',
+  downloadLink: 'www.google.com',
```
[the remark this answers is not in the document]
Comment (charisk) [reply extensions/ql-vscode/src/remote-queries/view/RemoteQueries.tsx:85]: Good idea! I'll update.

File: extensions/ql-vscode/src/view/octicons.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (shati-patel) [reply extensions/ql-vscode/src/view/octicons.tsx:23]: Just a link sounds good too!

File: extensions/ql-vscode/src/remote-queries/view/RemoteQueries.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (charisk) [reply extensions/ql-vscode/src/remote-queries/view/RemoteQueries.tsx]: Done

File: extensions/ql-vscode/src/remote-queries/view/RemoteQueries.tsx [hunk 3]
```diff
@@ -1,15 +1,112 @@
 import * as React from 'react';
 import * as Rdom from 'react-dom';
+import * as octicons from '../../view/octicons';
```
Comment (aeisenberg) [inline extensions/ql-vscode/src/remote-queries/view/RemoteQueries.tsx:3]: I know this is what we have been using for a while, but could you verify that the octicons look fine in both light and dark modes?

File: extensions/ql-vscode/src/remote-queries/view/remoteQueries.css [hunk 4]
```diff
@@ -1,5 +1,93 @@
...
+
+p {
+  margin: 0.5em 0 0.5em 0;
+}
+
+a {
+  text-decoration: none;
+}
+
+.download-link {
```
[the remark this answers is not in the document]
Comment (aeisenberg) [reply extensions/ql-vscode/src/remote-queries/view/remoteQueries.css:14]: I think it's best to keep using the prefixes. 

The css in the original view uses [BEM style css](http://getbem.com/introduction/) in order to avoid naming conflicts. I'm not a fan of this convention anymore since there are now some better solutions, but they require using preprocessors in different ways.

This would be a nice clean-up task, to convert to sass or something similar. But until then, I think we should stick with the current approach.

File: extensions/ql-vscode/src/remote-queries/view/RemoteQueries.tsx [hunk 5]
```diff
@@ -1,15 +1,112 @@
...
+    <span className="analysis-item">{props.nwo}</span>
+    <span className="analysis-item badge-container">
+      <span className="badge">{props.resultCount}</span>
+    </span>
+    <span className="analysis-item">
+      <a
+        className="download-link"
+        href={props.downloadLink}>
+        {octicons.download}{props.fileSize}
+      </a>
```
Comment (aeisenberg) [inline extensions/ql-vscode/src/remote-queries/view/RemoteQueries.tsx:37]: So, is the idea for the webview to make the requests directly? Or will it pass a message to the extension to make the request for the webview?

For now, this doesn't need to be worked out completely, but this is going to be an important decision to make. Some things to think about:

1. If the webview makes the request directly, we need to make sure the CSP allows it to happen (not difficult to do).
2. If the webview makes the request, then we need to figure out how to cache the data between runs. Does it send the data to the extension through a message and the extension saves it? Or does the webview save it directly (I think this is possible, but need to figure out exactly how the webview can access the file system).

My opinion is that _initially_ it will be simpler to make the request directly through the webview, but it will make things more complex later on since the webview still needs to communicate with the extension for caching and displaying query contents in the query history view.

Whatever decision we make initially, let's try to do it in a way that it's not difficult to change.

**Review by aeisenberg — COMMENTED** [review]
Nice work.

File: extensions/ql-vscode/src/remote-queries/view/RemoteQueries.tsx (hunk 5 above)
Comment (charisk) [reply extensions/ql-vscode/src/remote-queries/view/RemoteQueries.tsx:37]: In general, my thinking was that it would keep things simpler if the WebView would communicate with the rest of the world via the extension engine. But I completely forgot about it here, so you make a really good point!

I think we should leave things as is for now since this is a mock view, but when it comes to it, those links can turn into buttons that communicate with the extension that manages the download.

File: extensions/ql-vscode/src/remote-queries/view/remoteQueries.css (hunk 4 above)
[the remark this answers is not in the document]
Comment (charisk) [reply extensions/ql-vscode/src/remote-queries/view/remoteQueries.css]: Thank you both! I've made the changes to stick to the current convention (I think!) but I'm a bit unclear why the prefixes are needed (do all styles go into a global space, not encapsulated within the React component?).

Agree it'd be cool to start using something like LESS (which seems to be configured already, looking at webpack.config.ts) when we get the chance.

[the remark this answers is not in the document]
Comment (aeisenberg) [reply extensions/ql-vscode/src/remote-queries/view/remoteQueries.css]: I'm pretty sure (but haven't checked recently that all css is in the global namespace. Currently, each webview only has a single css file so it's not really a big deal in practice.

**Review by aeisenberg — APPROVED** [review]
This looks awesome!

**Comment by sj** [comment]
Looks smashing! Can I suggest a tiny change of wording to "Summary: 13 repositories affected". It's quite common for users to write queries that are meant to explore a code base. For such queries, the term "affected" might sound a bit odd. And the number "13" is already mentioned in the text just above it.

So how about rewording "Summary: 13 repositories affected" to "Repositories with results (13)" or something like that? @nuthinking and @jf205 might have better ideas!

**Comment by charisk** [comment]
> Looks smashing! Can I suggest a tiny change of wording to "Summary: 13 repositories affected". It's quite common for users to write queries that are meant to explore a code base. For such queries, the term "affected" might sound a bit odd. And the number "13" is already mentioned in the text just above it.
> 
> So how about rewording "Summary: 13 repositories affected" to "Repositories with results (13)" or something like that? @nuthinking and @jf205 might have better ideas!

Thanks @sj, I've gone ahead and made the change but we can also make further tweaks later on.

**Comment by nuthinking** [comment]
I know that we already show the number above in small but, also for consistency with the results header, I'd rather go for _"13 repositories with results"_ instead.
